ompi/mca/common/ompio/common_ompio_request.c

```diff
@@ -142,16 +150,66 @@ int mca_common_ompio_progress ( void )
         if( REQUEST_COMPLETE(&req->req_ompi) ) {
             continue;
         }
-        if ( NULL != req->req_progress_fn ) {
-            if ( req->req_progress_fn(req) ) {
-                completed++;
-                ompi_request_complete (&req->req_ompi, true);
+        if (NULL != req->req_progress_fn) {
```

Comment:
I believe the progress function can be called from multiple threads concurrently so there should be lock to only let one thread progress ompio and have others skip